### PR TITLE
FIX: prevents PM to large groups

### PR DESCRIPTION
**Pull request by jjaffeux**
This commit introduces a new hidden site setting: `group_pm_user_limit`, default to `1000` which will raise an error when attempting to create a PM target a large group.

### CODE REVIEW

File: lib/has_errors.rb [hunk 1]
```diff
@@ -15,14 +15,14 @@ def validate_child(obj)
...
+  def rollback_with!(obj, error, **kwargs)
+    obj.errors.add(:base, error, **kwargs)
     rollback_from_errors!(obj)
   end
 
   def rollback_from_errors!(obj)
     add_errors_from(obj)
-    raise ActiveRecord::Rollback.new
+    raise ActiveRecord::Rollback.new, obj.errors.full_messages.join("\n")
   end
```
Comment (jjaffeux) [inline lib/has_errors.rb:26]: I made these changes to allow tests to check the raised error message

File: lib/topic_creator.rb [hunk 2]
```diff
@@ -109,7 +109,13 @@ def watch_topic(topic)
...
 
     topic.reload.topic_allowed_groups.each do |topic_allowed_group|
-      topic_allowed_group.group.set_message_default_notification_levels!(topic)
+      group = topic_allowed_group.group
+
+      begin
+        group.set_message_default_notification_levels!(topic)
+      rescue Group::GroupPmUserLimitExceededError => e
+        rollback_with!(topic, :too_large_group, group_name: group.name)
+      end
```
Comment (jjaffeux) [inline lib/topic_creator.rb:118]: This is not the most elegant, but it seemed necessary to follow the same patterns used at other places in topic_creator

File: config/locales/server.en.yml [hunk 3]
```diff
@@ -612,6 +612,7 @@ en:
         one: "%{count} user has been added to the group."
         other: "%{count} users have been added to the group."
     errors:
+      default_notification_level_users_limit: "Default notification level can't be set on groups larger than %{count} users. Incorrect group: %{group_name}"
```
Comment (martin-brennan) [inline config/locales/server.en.yml:615]: We should have `one:` and `other:` for translations with count to cover the different singular and plural form of `user`

File: config/locales/server.en.yml [hunk 4]
```diff
@@ -754,6 +755,7 @@ en:
               reply_by_email_disabled: "Reply by email has been disabled."
               send_to_email_disabled: "Sorry, you cannot send personal messages to email."
               target_user_not_found: "One of the users you are sending this message to could not be found."
+              too_large_group: "The group: %{group_name}, has too many users to receive a message."
```
Comment (martin-brennan) [inline config/locales/server.en.yml:758]: This seems to be the actual error shown to the users but we don't have the max count in it, we should add it I think.

Comment (martin-brennan) [reply config/locales/server.en.yml:758]: But maybe it's not possible in topic creator? IDK...

**Review by martin-brennan — APPROVED** [review]
Approving, just have notes on translations

File: config/locales/server.en.yml (hunk 3 above)
Comment (jjaffeux) [reply config/locales/server.en.yml:615]: Yes I hesitated, my position is that a limit of 1/0 seems totally stupid, so this extra string seems useless 🤷‍♂️

File: config/locales/server.en.yml (hunk 4 above)
Comment (jjaffeux) [reply config/locales/server.en.yml:758]: It is possible since I added my changes, I thought it was maybe too much details for end user.

**Review by Flink — APPROVED** [review]

**Pull request merged**